http_check/datadog_checks/http_check/http_check.py

```diff
@@ -302,18 +329,27 @@ def check_cert_expiration(self, instance, timeout, instance_ca_certs):
             context.load_verify_locations(instance_ca_certs)
 
             ssl_sock = context.wrap_socket(sock, server_hostname=server_name)
-            cert = ssl_sock.getpeercert()
-            if cert:
-                exp_date = datetime.strptime(cert['notAfter'], "%b %d %H:%M:%S %Y %Z")
-            else:
+            binary_cert = ssl_sock.getpeercert(binary_form=True)
+
+            # To maintain backwards compatability, if we aren't validating tls/certs, do not process
+            # the returned binary cert unless specifically configured to with tls_retrieve_non_validated_cert
+            if (
+                instance.get("tls_verify") is False and not instance.get("tls_retrieve_non_validated_cert")
```

Comment:
```suggestion
                instance.get("tls_verify") is False and is_affirmative(instance.get("tls_retrieve_non_validated_cert", False))
```

Reply:
I committed this by an accident from this comment and am working on fixing it up now. What is the best way to import this function? I'm getting errors with the linux environment with import failures while attempting to import from `datadog_checks_base.datadog_checks.base`.

Reply:
Please disregard this...this was my error :)

Reply:
This has been updated and tests are passing now